src/priority_queue.js

```diff
@@ -0,0 +1,109 @@
+'use strict'
+
+class Node {
+  constructor(data, priority, next) {
+    this.data = data
+    this.priority = priority
+    this.next = next
+  }
+}
+
+export default class Queue {
+  constructor() {
+    this.thequeue = null
+    this.size = 0
+  }
+
+  enqueue(input, priority) {
+    if (!this.thequeue) {
+      this.thequeue = new Node(input, priority, null)
+      this.size++
+      return
+    }
+
+    let currentNode = this.thequeue
+    while (currentNode.next) {
+      currentNode = currentNode.next
+    }
+
+    currentNode.next = new Node(input, priority, null)
+
+    this.size++
+
+  }
+
+  front() {
+    if (!this.thequeue) {
+      return null
+    }
+    let evaluateHigherPriority = () => {
+      if (currentNode.priority > highestPriorityNode.priority) {
+        highestPriorityNode = currentNode
+      }
+    }
+    let highestPriorityNode = this.thequeue
+    let currentNode = this.thequeue
+    while (currentNode.next) {
+      evaluateHigherPriority()
+      currentNode = currentNode.next
+    }
+    evaluateHigherPriority()
+    return highestPriorityNode.data
+  }
+
+  back() {
+    if (!this.thequeue) {
+      return null
+    }
+    let evaluateLowerPriority = () => {
```

Comment:
i think this should be a const --- theres also a principle called single responsibility principle. you could define this function, ``` const evaluateLowerPriority = () => { ... } ``` outside of this function and refer to it inside of back()